### ISSUE TITLE
[microprofile] openapi response is wrong

### ISSUE THREAD

**Issue by Berger1994** [payara/Payara#3832]
# Description #
----------

I'm trying to learn the Microprofile: OpenAPI feature, but it seems that the generated openapi file for list response entities are wrong.

The attribute $ref is missing for array items

## Expected Outcome

    /resources/servers:
      get:
        operationId: getServers
        responses:
          default:
            description: Default Response.
            content:
              application/json:
                schema:
                  type: array
                  items: 
                    $ref: '#/components/schemas/Server'

## Current Outcome

    /resources/servers:
      get:
        operationId: getServers
        responses:
          default:
            description: Default Response.
            content:
              application/json:
                schema:
                  type: array

## Steps to reproduce (Only for bug reports) 

Create simple dto

```
@Schema(name = "Server")
public class Server {
	@Schema(name = "name", required = true)
	private String name;

	public Server(String name) {
		super();
		this.name = name;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}
}

```

Create simple resource
```
@Path("/servers")
public class ServerResource {

	private static final Server SERVER_A = new Server("Server A");
	private static final Server SERVER_B = new Server("Server B");
	private static final List<Server> SERVERS = Arrays.asList(SERVER_A, SERVER_B);

	@GET
	@Produces(MediaType.APPLICATION_JSON)
	public List<Server> getServers() {
		return SERVERS;
	}
}
```

```
@ApplicationPath("resources")
public class JAXRSConfiguration extends Application {

}

```

Start the payara, deploy the war and open the page [http://localhost:8080/openapi](http://localhost:8080/openapi)

## Context (Optional)

I'm trying to learn the Microprofile: OpenAPI.

## Environment ##

- **Payara Version**: 5.191
- **Edition**: Full
- **JDK Version**: 1.8.0_181
- **Operating System**:  Windows
- **Database**: not required

**Comment by OndroMih** [comment]
You are correct. The OpenAPI specification defines that "items MUST be present if the type is array."

**Comment by jbee** [comment]
I am currently working on updating from Open API 1.0 to 1.1 and we did find a few issues that were not according to spec. On first glance this does not strike me as one of those. In any case we should wait until 1.1 is complete and re-check if the issue is still present.

**Comment by jbee** [comment]
I just identified the problem. Both 1.0 and 1.1 TCK did not find this but serialisation of anything that is a `Map` is broken in the sense that only the map entries are rendered while other getters of the type (for example `getRef` or `getExtensions` are ignored).

**Comment by jbee** [comment]
Going to be fixed in connection with 1.1 update.

**Comment by jbee** [comment]
Reminder: There are multiple problems currently: 

1. `items` node is not rendered
2. `$ref` within might not be rendered if if `items` would be

**Comment by jbee** [comment]
This issue was on multiple levels:
1. not processing inner classes
2. JSON rendering wasn't always reflecting the model correctly

Fix will be in #3852 together with a test for the given example, look for `SchemaReferenceTest`.

**Comment by smillidge** [comment]
Fixed by #3852
